DCOS/templates/preprovision/extensions/preprovision-agent-windows/v1/preprovision-agent-windows.ps1

```diff
@@ -0,0 +1,12 @@
+New-Item -ItemType Directory -Force -Path C:/DCOS/mesos/service/
```

Comment:
I recommend having a variable:
```
$MESOS_SERVICE_DIR = Join-Path $env:SystemDrive "DCOS\mesos\service"
```
and use that when needed. For example when writing to file:
```
[System.IO.File]::WriteAllLines("$MESOS_SERVICE_DIR\credential.json", $cred, $UTF8NoBOM)
```
In case somehow this gets changed in the future, we only need to change a single variable.

Reply:
Thanks Ionut, good point, I'll make the change